README.md

```diff
@@ -11,51 +11,61 @@ related-issues: (will contain links to implementation PRs)
+2. Are defined to be equal to `pkgs.callPackage <path> { }` such that all transitively referenced paths from the default Nix file of `<path>`:
+  1. Are under the directory of `<path>`
+    (necessary so that moving these files to a new directory doesn't break references in this package)
+  2. Are not referenced from any paths outside of these transitive references
+    (necessary so that moving these files to a new directory doesn't break references in other packages)
+3. Evaluate to a derivation
+  (necessary because using `pkg-fun.nix` for a non-package would be counter-intuitive)
 
-```nix
-{
-  <name> = pkgs.callPackage ../some/dir { };
-}
-```
+These will be become eligible to be transformed as follows:
+- Move the default Nix file from `<path>` to `pkgs/unit/<4-prefix name>/<name>/pkg-fun.nix` ([TODO: Justify why `unit`](https://github.com/nixpkgs-architecture/simple-package-paths/issues/16))
+  - Where `<4-prefix name>` is the 4-letter prefix of `<name>`.
+    If `<name>` has less than 4 characters, append `-` to the `<name>` until it's 4 characters long and use that as `<4-prefix name>`
+- Additionally also move all paths transitively referenced by the default Nix file to `pkgs/unit/<4-prefix name>/<name>/???` [TODO](https://github.com/nixpkgs-architecture/simple-package-paths/issues/19)
+- Remove the definition of that attribute in `pkgs/top-level/all-packages.nix`
+- For each moved path, create a compatibility layer from the old to the new path, potentially using a symlink. See [compatibility] for more details
 
-will be become eligible to be transformed as follows:
-- Move `pkgs/some/dir` to `pkgs/auto/<name>`
-- Fix any references to that folder (e.g. in update scripts) 
-- Remove the original definition
+These attributes will newly be added to `pkgs` by automatically calling `pkgs.callPackage pkgs/unit/<4-prefix name>/<name>/pkg-fun.nix { }` on all entries in `pkgs/unit`. In order to make this more efficient, `builtins.readDir` should be optimized as described [here](https://github.com/NixOS/nix/issues/7314).
 
-These attributes will newly be defined by listing all directories in `pkgs/auto` using `builtins.readDir` and calling `pkgs.callPackage` on all of them, which then gets added to the `pkgs` scope.
+## Compatibility layer
+[compatibility]: #compatibility-layer
+
+TODO: Nix files should use `import` to act like a symlink while also giving a warning with `builtins.trace`. Something like
 
-Attributes whose definition aren't of the above form won't be changed, so e.g. the following definition in `all-packages.nix` won't be changed:
 ```nix
-{
-  syncplay-nogui = syncplay.override { enableGUI = false; };
-}
+builtins.trace "warning: Using deprecated path ${./.}, use pkgs/unit/<name> instead, this will be removed after NixOS 22.05"
+  (import ../../pkgs/unit/name)
```

Comment:
That does not work with ofborg and potentially spams people evaluation log.

Reply:
No it shouldn't, since nothing in nixpkgs evaluates these files anymore at that point.